### PR TITLE
Improve test isolation in DeprecationHttpIT based on xOpaqueId

### PR DESCRIPTION
**Pull request by mosche**
Rather than trying extremely hard to pretend isolation on a shared resource (the deprecation index), this instead attempts to provide isolation by strictly using a unique xOpaqueId per test case.

This should even allow to preserve cluster state between runs:
```
    @Override
    protected boolean preserveClusterUponCompletion() {
        return true; // isolation is based on xOpaqueId
    }
```

Additionally this sets the flush interval of the deprecation bulk processor to achieve reasonably fast test runs.

Also, on the positive side, this makes the test much simpler again and allows to focus primarily on the functional aspects.

### CODE REVIEW

**Review by mark-vieira — APPROVED** [review]

**Review by williamrandolph — APPROVED** [review]
This is a cool fix. LGTM.

File: x-pack/plugin/deprecation/qa/rest/src/javaRestTest/java/org/elasticsearch/xpack/deprecation/DeprecationHttpIT.java [hunk 1]
```diff
@@ -260,7 +218,7 @@ public void testDeprecationHeadersDoNotGetStuck() throws Exception {
      * <p>
      * Re-running this back-to-back helps to ensure that warnings are not being maintained across requests.
```
[the remark this answers is not in the document]
Comment (mosche) [reply x-pack/plugin/deprecation/qa/rest/src/javaRestTest/java/org/elasticsearch/xpack/deprecation/DeprecationHttpIT.java:219]: This is still valid, this is repeated several times in a single test case.

File: x-pack/plugin/deprecation/qa/common/src/main/java/org/elasticsearch/xpack/deprecation/DeprecationTestUtils.java [hunk 2]
```diff
@@ -17,18 +17,25 @@
...
+    }
+
+    @SuppressWarnings("unchecked")
+    static List<Map<String, Object>> getIndexedDeprecations(RestClient client, String xOpaqueId) throws IOException {
         Response response;
         try {
             client.performRequest(new Request("POST", "/" + DATA_STREAM_NAME + "/_refresh?ignore_unavailable=true"));
-            response = client.performRequest(new Request("GET", "/" + DATA_STREAM_NAME + "/_search"));
+            String query = xOpaqueId == null ? "" : "?q=" + X_OPAQUE_ID_FIELD_NAME + ":" + xOpaqueId;
+            response = client.performRequest(new Request("GET", "/" + DATA_STREAM_NAME + "/_search" + query));
```
[the remark this answers is not in the document]
Comment (mosche) [reply x-pack/plugin/deprecation/qa/common/src/main/java/org/elasticsearch/xpack/deprecation/DeprecationTestUtils.java:38]: Not sure I understand? Why a header?
This is defining a search query on the deprecation index using on of the different possible ways to do so

[the remark this answers is not in the document]
Comment (pgomulka) [reply x-pack/plugin/deprecation/qa/common/src/main/java/org/elasticsearch/xpack/deprecation/DeprecationTestUtils.java:38]: ah fair point, I thought that this is triggering a deprecation.
all clear :) thank you

**Pull request merged**